BardMusicPlayer.Notate/Processor/AutoToneProcessor.cs

```diff
@@ -0,0 +1,85 @@
+                Config.AutoToneOctaveRange.LowerNote,
+                Config.AutoToneOctaveRange.LowerNote + Config.AutoToneInstrumentGroup.Size1 * 12 - 1,
+                (int) Config.AutoToneInstrumentGroup.Instrument1.InstrumentToneMenuKey,
+                false,
+                -OctaveRange.Parse(Config.AutoToneOctaveRange.Index).LowerNote)
+                .MoveNoteDictionaryToDefaultOctave(OctaveRange.Parse(Config.AutoToneOctaveRange.Index))
+                .ConcatNoteDictionaryToList();
+
+            var instrumentGroup2 = trackChunks.GetNoteDictionary(Song.SourceTempoMap, Config.AutoToneInstrumentGroup.InstrumentTone,
+                Config.AutoToneOctaveRange.LowerNote + Config.AutoToneInstrumentGroup.Size1 * 12,
+                Config.AutoToneOctaveRange.LowerNote + Config.AutoToneInstrumentGroup.Size1 * 12 + Config.AutoToneInstrumentGroup.Size2 * 12 - 1,
+                (int) Config.AutoToneInstrumentGroup.Instrument2.InstrumentToneMenuKey,
+                false,
+                -OctaveRange.Parse(Config.AutoToneOctaveRange.Index + 1).LowerNote)
+                .MoveNoteDictionaryToDefaultOctave(OctaveRange.Parse(Config.AutoToneOctaveRange.Index + 1))
+                .ConcatNoteDictionaryToList();
+
+            var instrumentGroup3 = trackChunks.GetNoteDictionary(Song.SourceTempoMap, Config.AutoToneInstrumentGroup.InstrumentTone,
+                Config.AutoToneOctaveRange.LowerNote + Config.AutoToneInstrumentGroup.Size1 * 12 + Config.AutoToneInstrumentGroup.Size2 * 12,
+                Config.AutoToneOctaveRange.LowerNote + Config.AutoToneInstrumentGroup.Size1 * 12 + Config.AutoToneInstrumentGroup.Size2 * 12 + Config.AutoToneInstrumentGroup.Size3 * 12,
+                (int) Config.AutoToneInstrumentGroup.Instrument3.InstrumentToneMenuKey,
+                false,
+                -OctaveRange.Parse(Config.AutoToneOctaveRange.Index + 2).LowerNote)
+                .MoveNoteDictionaryToDefaultOctave(OctaveRange.Parse(Config.AutoToneOctaveRange.Index + 2))
+                .ConcatNoteDictionaryToList();
+
+            await Task.WhenAll(instrumentGroup1, instrumentGroup2, instrumentGroup3);
+
+            var notes = new List<Note>();
+
+            if (!Config.AutoToneInstrumentGroup.Instrument1.Equals(Instrument.None)) notes.AddRange(instrumentGroup1.Result);
+            if (!Config.AutoToneInstrumentGroup.Instrument2.Equals(Instrument.None)) notes.AddRange(instrumentGroup2.Result);
+            if (!Config.AutoToneInstrumentGroup.Instrument3.Equals(Instrument.None)) notes.AddRange(instrumentGroup3.Result);
+            
+            var trackChunk = TimedObjectUtilities.ToTrackChunk(notes);
+
+            var playerNotesDictionary = await trackChunk.GetPlayerNoteDictionary(Config.PlayerCount, OctaveRange.C3toC6.LowerNote, OctaveRange.C3toC6.UpperNote);
+
+            var concurrentPlayerTrackDictionary = new ConcurrentDictionary<long, TrackChunk>();
```

Comment:
this should be preallocated for speed gains / step performance when populated in the parallel loop below